### ISSUE TITLE
Holding Backspace generates text on Android occasionally

### ISSUE THREAD

**Issue by hesamat** [codemirror/CodeMirror#5245]
Repro steps:
1- Open the [selection marking demo](https://codemirror.net/demo/markselection.html) page in Chrome on Android. (I am testing on Android 7.1.1, Chrome 64).
2- Hold backspace to delete characters.

Result:
Observe how every now and then some text gets added before getting deleted. In my project, the situation is worse as the text keeps getting duplicated :( 
For example, if you are deleting "Little", holding down Backspace would result in something like this: LittLittLittLitt

I looked into the pollContent method in ContentEditableInput module to see if there is a way to prevent text expansion in case we are deleting text, but couldn't figure how to detect deletion.

**Comment by marijnh** [comment]
Not really. The input system probably gets confused when reading new text content from the DOM, after a composition event (which is how Android models backspace), but I don't know why that'd be happening.

**Comment by edemaine** [comment]
@hesamat Which keyboard are you using?  I just tested on SwiftKey and Gboard, and it worked fine.  Or are you talking about an actual physical/emulated keyboard?

**Comment by hesamat** [comment]
@edemaine  I was using Gboard I believe. It was happening sporadically for me.

**Comment by adrianheine** [comment]
This is one of many issues which are difficult to solve or even debug with the fundamental approach currently taken by CodeMirror.

We are working on a rewrite (CodeMirror 6) that will overhaul input handling and should address this issue, and we are currently raising money for this work: See [the announcement](https://codemirror.net/6/) for more information about the rewrite and a demo.

Note that CodeMirror 6 is by no means stable or usable in production, yet. It is highly unlikely that we pick up this issue for CodeMirror 5, though.
